wire-ios-data-model/Source/Repositories/FeatureRepository.swift

```diff
@@ -277,23 +278,38 @@ public class FeatureRepository: FeatureRepositoryInterface {
 
     // MARK: - MLS
 
+    public func fetchMLS() async -> Feature.MLS {
+        let (status, configData) = await context.perform {
+            let feature = Feature.fetch(name: .mls, context: self.context)
+            return (feature?.status, feature?.config)
+        }
+
+        return makeMLS(status: status, configData: configData)
+    }
+
     public func fetchMLS() -> Feature.MLS {
-        guard
-            let feature = Feature.fetch(name: .mls, context: context),
-            let featureConfig = feature.config
-        else {
+        let (status, configData) = context.performAndWait {
+            let feature = Feature.fetch(name: .mls, context: context)
+            return (feature?.status, feature?.config)
+        }
```

Comment:
I needed to avoid crashes in `fetchMLS()` by using the context, but the outside calling `GetMLSFeatureUseCase` has no idea about contexts.
In my opinion we should have the line of abstraction here and thus I decided to call context.perform here either sync or async.